README.md

```diff
@@ -1,25 +1,71 @@
-# WSO2 Extension Siddhi Execution Priority
+siddhi-execution-priority
+======================================
 
+The **siddhi-execution-priority extension** is an extension to <a target="_blank" href="https://wso2.github.io/siddhi">Siddhi</a> that keeps track of the priority of events in a stream.
 
+Find some useful links below:
 
- PriorityStreamProcessor keeps track of the priority of events in a stream. This stream processor requires three arguments which are:
- 
- - Unique key variable to identify the event
- - Priority variable which contains the priority increment
- - Timeout in constant to decrease the priority by one after the given timeout
- 
+* <a target="_blank" href="https://github.com/wso2-extensions/siddhi-execution-priority">Source code</a>
+* <a target="_blank" href="https://github.com/wso2-extensions/siddhi-execution-priority/releases">Releases</a>
+* <a target="_blank" href="https://github.com/wso2-extensions/siddhi-execution-priority/issues">Issue tracker</a>
+
+## Latest API Docs 
+
+Latest API Docs is <a target="_blank" href="https://wso2-extensions.github.io/siddhi-execution-priority/api/4.0.2-SNAPSHOT">4.0.2-SNAPSHOT</a>.
+
+## How to use 
+
+**Using the extension in <a target="_blank" href="https://github.com/wso2/product-sp">WSO2 Stream Processor</a>**
+
+* You can use this extension in the latest <a target="_blank" href="https://github.com/wso2/product-sp/releases">WSO2 Stream Processor</a> that is a part of <a target="_blank" href="http://wso2.com/analytics?utm_source=gitanalytics&utm_campaign=gitanalytics_Jul17">WSO2 Analytics</a> offering, with editor, debugger and simulation support. 
+
+* This extension is shipped by default with WSO2 Stream Processor, if you wish to use an alternative version of this extension you can replace the component <a target="_blank" href="https://github.com/wso2-extensions/siddhi-execution-priority/releases">jar</a> that can be found in the `<STREAM_PROCESSOR_HOME>/lib` directory.
+
+**Using the extension as a <a target="_blank" href="https://wso2.github.io/siddhi/documentation/running-as-a-java-library">java library</a>**
+
+* This extension can be added as a maven dependency along with other Siddhi dependencies to your project.
+
+```
+     <dependency>
+        <groupId>org.wso2.extension.siddhi.execution.priority</groupId>
```

Comment:
We should not add the parent ad the dependency should use the component

Reply:
Fixed in b6f023e commit.